src/posts/lucky_0.29.cr

```diff
@@ -0,0 +1,192 @@
+    in your `config/database.cr` by changing the `query_cache_enabled` setting accordingly.
+
+    The query cache mechanism is designed to store your ModelQuery results in to memory
+    over the duration of an HTTP request.
+
+    ```crystal
+    # this hits the database
+    UserQuery.new.first
+
+    # this is pulled from cache in-memory
+    UserQuery.new.first
+    ```
+
+    ### More with validations
+
+    We've also made some improvements with validations. Each validation method
+    now returns a `Bool` value if it succeeded or not.
+
+    ```crystal
+    before_save do
+      if validate_required(code)
+        run_super_complex_validation
+      end
+    end
+    ```
+
+    There's also a new validation method `validate_format_of`. Use this validation
+    to test if an attribute meets a specific format.
+
+    ```crystal
+    before_save do
+      validate_format_of pin, with: /^d+$/
+    end
+    ```
+
+    As you may know, non-nilable attributes are automatically ran through
+    required validations. Since they can't be nil, we need to ensure they
+    have a value. To go along with these default validations, we also have
+    a new macro called `default_validations`.
+
+    The `default_validations` macro allows you to specify your own custom
+    default validations that you need to run. These validations are ran
+    after all of the `before_save` blocks have completed allowing you to
+    do any necessary setup before hand.
+
+    ```crystal
+    before_save do
+      code.value ||= "1234"
+    end
+
+    default_validations do
+      validate_required code
+    end
+    ```
+
+    In some instances, you may need to just skip all of the default validations
+    , including the ones Lucky runs for you. To do this, we've added the `skip_default_validations`
+    macro. Place this in the operation you wish to skip those validations.
+
+    ```crystal
+    class SaveCode < Code::SaveOperation
+      skip_default_validations
+
+    end
+    ```
+
+    Lastly, the validations are all now compatible with I18n. You have many choices
+    for shards that can handle translations for you. Create a struct and inherit from
+    `Avram::I18nBackend` to override the defaults.
+
+    ```crystal
+    # config/i18n.cr
+    struct I18n < Avram::I18nBackend
+      def get(key : String | Symbol) : String
+        # do your lookup here
+      end
+    end
+
+    # config/database.cr
+    Avram.configure do |settings|
+      # ...
+
+      settings.i18n_backend = I18n.new
+    end
+    ```
+
+    > [Read the PR](https://github.com/luckyframework/avram/pull/757) for more details.
+
+    ## Parting words
+
+    We want to acknowledge and thank the community members for sticking with the Lucky framework.
+    Lucky has set a goal to be a solid and stable choice for a Crystal based web-framework and in
+    most cases, we feel we have hit the mark.
+
+    At this point we had hoped to be closer to the 1.0 release, but due to life outside of OpenSource,
```

Comment:
We're not very consistent about this but I don't think [open source](https://en.wiktionary.org/wiki/open_source) is a proper noun. Also, confusingly, when it's an [adjective](https://en.wiktionary.org/wiki/open-source) (e.g. open-source software) it should use the hyphen. We've been pretty good about that one so far at least in blog posts.

```suggestion
    At this point we had hoped to be closer to the 1.0 release, but due to life outside of open source,
```